cmd/ndm_daemonset/probe/sysfsprobe.go

```diff
@@ -0,0 +1,155 @@
+	sysfsProbeName  = "sysfs probe"
+	sysfsProbeState = defaultEnabled
+)
+
+var sysfsProbeRegister = func() {
+	ctrl := <-controller.ControllerBroadcastChannel
+	if ctrl == nil {
+		klog.Error("unable to configure", sysfsProbeName)
+		return
+	}
+	if ctrl.NDMConfig != nil {
+		for _, probeConfig := range ctrl.NDMConfig.ProbeConfigs {
+			if probeConfig.Key == sysfsConfigKey {
+				sysfsProbeName = probeConfig.Name
+				sysfsProbeState = util.CheckTruthy(probeConfig.State)
+				break
+			}
+		}
+	}
+
+	newRegistryProbe := &registerProbe{
+		priority:   sysfsProbePriority,
+		name:       sysfsProbeName,
+		state:      sysfsProbeState,
+		pi:         newSysFSProbe(),
+		controller: ctrl,
+	}
+	newRegistryProbe.register()
+}
+
+// sysfsProbe fills the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+type sysfsProbe struct {
+}
+
+func newSysFSProbe() *sysfsProbe {
+	return &sysfsProbe{}
+}
+
+// It is part of probe interface. Hence, empty implementation.
+func (cp *sysfsProbe) Start() {}
+
+// FillBlockDeviceDetails updates the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+// if those are not populated.
+func (cp *sysfsProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDevice) {
+	sysPath := blockDevice.SysPath
+
+	if blockDevice.DeviceAttributes.LogicalBlockSize == 0 {
+		logicalBlockSize, err := readSysFSFileAsInt64(sysPath + "/queue/logical_block_size")
+		if err != nil {
+			klog.Warning("unable to read logical block size", err)
+		} else if logicalBlockSize != 0 {
+			blockDevice.DeviceAttributes.LogicalBlockSize = uint32(logicalBlockSize)
+			klog.Infof("blockdevice path: %s logical sector size :%d filled by queuesysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.LogicalBlockSize)
+		}
+	}
+
+	if blockDevice.DeviceAttributes.PhysicalBlockSize == 0 {
+		physicalBlockSize, err := readSysFSFileAsInt64(sysPath + "/queue/physical_block_size")
+		if err != nil {
+			klog.Warning("unable to read physical block size", err)
+		} else if physicalBlockSize != 0 {
+			blockDevice.DeviceAttributes.PhysicalBlockSize = uint32(physicalBlockSize)
+			klog.Infof("blockdevice path: %s physical sector size :%d filled by sysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.PhysicalBlockSize)
+		}
+	}
+
+	if blockDevice.DeviceAttributes.DriveType == "" {
+		rotational, err := readSysFSFileAsInt64(sysPath + "/queue/rotational")
+		if err != nil {
+			klog.Warning("unable to read rotational value", err)
+		} else {
+			if rotational == 1 {
+				blockDevice.DeviceAttributes.DriveType = "HDD"
+
+			} else if rotational == 0 {
+				blockDevice.DeviceAttributes.DriveType = "SSD"
+			}
+			klog.Infof("blockdevice path: %s drive type :%s filled by sysfs probe.",
+				blockDevice.DevPath, blockDevice.DeviceAttributes.DriveType)
+		}
+	}
+
+	if blockDevice.Capacity.Storage == 0 {
+		hwSectorSize, err := readSysFSFileAsInt64(sysPath + "/queue/hw_sector_size")
```

Comment:
These can be replaced with as mentioned in `capacity-probe` of PR #378.

Reply:
ok.
You will send me a patch or can I can copy the changes from there?
I am ok with both.

Reply:
Ok got the answer in another PR thank you. I will add you as a co-author.